### PR TITLE
Update bundler to 2.2.33

### LINKED ISSUE #1225: Avoid deployments issues caused by bundler with git sources using globs
Hello,

The current bundler version has an [issue](https://github.com/rubygems/rubygems/issues/4943) linked to the generation of Gemfile.lock with multiple git sources on the same repository, differentiated by their glob: their order generated in the Gemfile.lock is non-deterministic and differ between operating systems.

<details>

<summary> This difference causes the deployment on a dyno to fail </summary>

```
Installing dependencies using bundler 2.2.21        
remote:        Running: BUNDLE_WITHOUT='development:test' BUNDLE_PATH=vendor/bundle BUNDLE_BIN=vendor/bundle/bin BUNDLE_DEPLOYMENT=1 bundle install -j4        
remote:        You are trying to install in deployment mode after changing        
remote:        your Gemfile. Run `bundle install` elsewhere and add the        
remote:        updated Gemfile.lock to version control.        
remote:                
remote:        If this is a development machine, remove the /tmp/build_349b20f0/Gemfile freeze         
remote:        by running ``.        
remote:                
remote:        The list of sources changed        
remote:        Bundler Output: You are trying to install in deployment mode after changing        
remote:        your Gemfile. Run `bundle install` elsewhere and add the        
remote:        updated Gemfile.lock to version control.        
remote:                
remote:        If this is a development machine, remove the /tmp/build_349b20f0/Gemfile freeze         
remote:        by running ``.        
remote:                
remote:        The list of sources changed        
remote:         
remote:  !        
remote:  !     Failed to install gems via Bundler.        
remote:  !        
remote:  !     Push rejected, failed to compile Multipack app.        
remote: 
remote:  !     Push failed  
```

</details>

This is fixed in [v2.2.29](https://github.com/rubygems/rubygems/releases/tag/bundler-v2.2.29) by https://github.com/rubygems/rubygems/pull/4947

As the heroku buildpack still uses a bundler version that precedes the version with this fix, we are having problems deploying properly to our production dynos. Could it be possible to bump the bundler version used to solve this problem?

### PR DESCRIPTION
**Pull request by schneems**
close #1225

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## Main (unreleased)
 
+* Bundler 2.x is now 2.2.33 ()
```
Comment (edmorley) [inline CHANGELOG.md:5]:
```suggestion
* Bundler 2.x is now 2.2.33 (https://github.com/heroku/heroku-buildpack-ruby/pull/1248)
```

File: changelogs/unreleased/bundler_2_2_33.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+##  Ruby apps with Bundler 2.x now receive version 2.2.33
+
+The [Ruby Buildpack](https://devcenter.heroku.com/articles/ruby-support#libraries) now includes Bundler 2.2.11. Applications specifying Bundler 2.x in their `Gemfile.lock` will now receive this version of bundler.
```
Comment (edmorley) [inline changelogs/unreleased/bundler_2_2_33.md:3]:
```suggestion
The [Ruby Buildpack](https://devcenter.heroku.com/articles/ruby-support#libraries) now includes Bundler 2.2.33. Applications specifying Bundler 2.x in their `Gemfile.lock` will now receive this version of bundler.
```

**Review by edmorley — APPROVED** [review]

**Comment by schinery** [comment]
@schneems are you able to @edmorley's PR suggestions done so this could potentially get merged? 🙏🏻

**Comment by schneems** [comment]
This is released in https://devcenter.heroku.com/changelog-items/2312

**Pull request merged**